istioctl/cmd/analyze.go

```diff
@@ -31,6 +34,15 @@ import (
 var (
 	useKube      bool
 	useDiscovery string
+	colorize     bool
+
+	colorEnvVar = env.RegisterBoolVar("COLORIZE_ISTIOCTL", true, "Color output (some commands)")
+
+	colorPrefixes = map[diag.Level]string{
+		diag.Info:    "",           // no special color for info messages
+		diag.Warning: "\033[33m",   // yellow
+		diag.Error:   "\033[1;31m", // bold red
```

Comment:
do we support windows? and if so does this work on Windows?

Reply:
I can ask around to see if anyone can test on Windows.  However, color capability will be very user-dependent.

Many Windows users will have the Windows 10 coloring turned on, or be using a shell that provides it.  It would be great if _istioctl_ could auto-detect if color was disabled and make the default `--color=false` in that case.  I don't know how to do that.  I could disable by default on Windows and require the user to explicitly enable coloring on Windows.

See https://github.com/ytdl-org/youtube-dl/issues/15758 for the best discussion I've found.

Reply:
re: auto-detecting color support, see https://github.com/fatih/color/blob/3f9d52f7176a6927daacff70a3e8d1dc2025c53e/color.go#L15-L21

Reply:
Marking this as `hold` so I can investigate this approach.

Reply:
@ayj @howardjohn in this PR I define `$ISTIOCTL_COLOR`.  I would be happy to use the more standard `$TERM`.
This code makes use of github.com/mattn/go-isatty ; do you think it would be reasonable to require it for istioctl?

Reply:
SGTM